sky/clouds/scp.py

```diff
@@ -0,0 +1,332 @@
+                memory=resources.memory,
+                disk_tier=resources.disk_tier)
+            if default_instance_type is None:
+                return ([], [])
+            else:
+                return (_make([default_instance_type]), [])
+
+        assert len(accelerators) == 1, resources
+        acc, acc_count = list(accelerators.items())[0]
+        (instance_list, fuzzy_candidate_list
+        ) = service_catalog.get_instance_type_for_accelerator(
+            acc,
+            acc_count,
+            use_spot=resources.use_spot,
+            cpus=resources.cpus,
+            memory=resources.memory,
+            region=resources.region,
+            zone=resources.zone,
+            clouds='scp')
+        if instance_list is None:
+            return ([], fuzzy_candidate_list)
+        return (_make(instance_list), fuzzy_candidate_list)
+
+    @classmethod
+    def check_credentials(cls) -> Tuple[bool, Optional[str]]:
+        try:
+            scp_utils.SCPClient().list_instances()
+        except (AssertionError, KeyError, scp_utils.SCPClientError):
+            return False, ('Failed to access SCP with credentials. '
+                           'To configure credentials, go to:\n'
+                           ' https://cloud.samsungsds.com/openapiguide\n'
+                           ' to generate API key and add the line\n'
+                           ' access_key = [YOUR API ACCESS KEY]\n'
+                           ' secret_key = [YOUR API SECRET KEY]\n'
+                           ' project_id = [YOUR PROJECT ID]\n'
+                           ' to ~/.scp/scp_credential')
+
+        return True, None
+
+    def get_credential_file_mounts(self) -> Dict[str, str]:
+        return {
+            f'~/.scp/{filename}': f'~/.scp/{filename}'
+            for filename in _CREDENTIAL_FILES
+        }
+
+    @classmethod
+    def get_current_user_identity(cls) -> Optional[List[str]]:
+        # TODO(jgoo1): Implement get_current_user_identity for SCP
+        return None
+
+    def instance_type_exists(self, instance_type: str) -> bool:
+        return service_catalog.instance_type_exists(instance_type, 'scp')
+
+    def validate_region_zone(self, region: Optional[str], zone: Optional[str]):
+        return service_catalog.validate_region_zone(region, zone, clouds='scp')
+
+    def accelerator_in_region_or_zone(self,
+                                      accelerator: str,
+                                      acc_count: int,
+                                      region: Optional[str] = None,
+                                      zone: Optional[str] = None) -> bool:
+        return service_catalog.accelerator_in_region_or_zone(
+            accelerator, acc_count, region, zone, 'scp')
+
+    @staticmethod
+    def is_disk_size_allowed(resources):
+        if resources.disk_size and \
+               (resources.disk_size < _SCP_MIN_DISK_SIZE_GB or
+                resources.disk_size > _SCP_MAX_DISK_SIZE_GB):
```

Comment:
nit: use parentheses instead of `\` for the code style.
```suggestion
        if (resources.disk_size and
               (resources.disk_size < _SCP_MIN_DISK_SIZE_GB or
                resources.disk_size > _SCP_MAX_DISK_SIZE_GB)):
```